did_doc_builder/src/schema/service.rs

```diff
@@ -0,0 +1,180 @@
+use std::collections::{HashMap, HashSet};
+
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+use crate::error::DIDDocumentBuilderError;
+
+use super::{types::uri::Uri, utils::OneOrList};
+
+type ServiceTypeAlias = OneOrList<String>;
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct Service {
+    id: Uri,
+    r#type: ServiceTypeAlias,
+    service_endpoint: String,
+    #[serde(flatten)]
+    #[serde(skip_serializing_if = "HashMap::is_empty")]
+    #[serde(default)]
+    extra: HashMap<String, Value>,
```

Comment:
I can answer this one

https://www.w3.org/TR/did-core/#services

> Each service extension MAY include additional properties and MAY further restrict the properties associated with the extension.

and so for example here
https://sovrin-foundation.github.io/sovrin/spec/did-method-spec-template.html#crud-operation-definitions
under the `Resolver DID Document Format`
there's DID Doc example which contains `routingKeys` and `recipientKeys` which are not part of w3c spec.

We are currently using, and will be using for a while more `did:sov` method.

Reply:
As discussed, we might be able to substitute this with some generic type with well defined fields, but it might not be worth the investment right now since we don't even know whether we can pull it off for all did methods. I understand what this is about now, so thanks 😄